app/src/main/java/space/taran/arkmemo/data/repositories/TextNotesRepository.kt

```diff
@@ -17,69 +20,156 @@ import kotlin.io.path.getLastModifiedTime
+        removeFileFromMemory(versionPath)
+        //we should delete all notes listed in version that don't have been forked? by now it just deletes the last note.
+    }
+
+    fun getAllNotesWithHistory(): List<TextNote> {
+        val versions = getAllVersions()
+        val notes = mutableListOf<TextNote>()
+        for (ver in versions) {
+            val rootResourceId = ver.meta!!.rootResourceId
+            val lastChildId = getLastChildrenId(rootResourceId)
+            val note = getNote(lastChildId)
+            notes.add(note)
+        }
+        return notes
     }
 
-    fun getAllNotes(): List<TextNote> {
+    fun getAllNotesFromVersion(rootResourceId: ResourceId): List<TextNote> {
+        val versionStorage = VersionStorage(getPath()!!)
+        val versions = versionStorage.versions(rootResourceId)
         val notes = mutableListOf<TextNote>()
+        val rootNote = getNote(rootResourceId)
+        notes.add(rootNote)
+        for (noteResId in versions) {
+            val note = getNote(noteResId)
+            notes.add(note)
+        }
+        return notes
+    }
+
+    fun getAllVersions(): List<Version> {
+        val versions = mutableListOf<Version>()//content
         val path = getPath()
-        var number = 0
         if (path != null) {
-            Files.list(path).forEach { filePath ->
-                if (filePath.fileName.extension == NOTE_EXT) {
-                    number += 1
+            val versionDir = File(path.toFile(), ARK_VERSIONS_DIR)
+            if (versionDir.exists() && versionDir.isDirectory) {
+                Files.list(versionDir.toPath()).forEach { filePath ->
                     try {
-                        val jsonFile = filePath.toFile()
-                        val fileReader = FileReader(jsonFile)
-                        val bufferedReader = BufferedReader(fileReader)
-                        val jsonTextNote = StringBuilder()
-                        with(bufferedReader) {
-                            forEachLine {
-                                jsonTextNote.append(it)
-                            }
-                            val content = JsonParser.parseNoteFromJson(jsonTextNote.toString())
-                            val size = Files.size(filePath)
-                            val id = computeId(size, filePath)
-                            val meta = ResourceMeta(
-                                id,
-                                filePath.fileName.toString(),
-                                filePath.extension,
-                                filePath.getLastModifiedTime(),
-                                size
-                            )
-
-                            val note = TextNote(content, meta)
-                            notes.add(note)
-                            close()
-                        }
+                        val fileName = filePath.fileName.toString()
+                        val rootResourceId: ResourceId = fileName.toLong()
```

Comment:
_(Future work)_

We have a method for creating filenames from ids:
https://github.com/ARK-Builders/arklib/blob/main/src/id.rs#L31

Example: https://github.com/ARK-Builders/arklib-android/blob/3334850384f84fe8d5f0545a221cc1fbc14e722e/arklib/src/lib.rs#L269